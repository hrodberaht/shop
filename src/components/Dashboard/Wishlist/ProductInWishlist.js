import React, { Component } from 'react';
import PropTypes from 'prop-types';
import AddToCartButton from '../../shared/AddToCartButton';

export default class ProductInWishlist extends Component {
  handleClick = () => {
    const {
      remove,
      product: { productId },
    } = this.props;
    remove(productId);
  };

  render() {
    const {
      product,
      product: {
        name, price, pcsOrder, totalPrice,
      },
      handleClickToCart,
    } = this.props;
    return (
      <tr>
        <td>{name}</td>
        <td>{price}</td>
        <td>{pcsOrder}</td>
        <td>
          <span>$</span>
          {totalPrice}
        </td>
        <td>
          <button type="button" className="btn btn-danger" onClick={this.handleClick}>
            X
          </button>
        </td>
        <td>
          <AddToCartButton
            className="btn btn-primary"
            product={product}
            handleClickToCart={handleClickToCart}
          />
        </td>
      </tr>
    );
  }
}

ProductInWishlist.propTypes = {
  product: PropTypes.shape({
    name: PropTypes.string,
    type: PropTypes.string,
    price: PropTypes.number,
  }).isRequired,
  remove: PropTypes.func.isRequired,
  handleClickToCart: PropTypes.func.isRequired,
};
