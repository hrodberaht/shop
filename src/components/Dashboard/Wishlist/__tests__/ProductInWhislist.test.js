import ProductInWishlist from '../ProductInWishlist';

const defaultProps = {
  product: {
    productId: 'product-1',
    type: 'Printer',
    name: 'ZX3',
    price: 400,
    inStock: '30',
  },
  remove: jest.fn(),
  handleClickToCart: jest.fn(),
};

const setup = buildSetup(ProductInWishlist, defaultProps);

describe('ProductInWishlist />', () => {
  it('should render without crash', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });
  it('should call remove after click', () => {
    const {
      wrapper,
      props: {
        product: { productId },
      },
    } = setup();
    wrapper.find('button').simulate('click');
    expect(wrapper.instance().props.remove).toHaveBeenCalledWith(productId);
  });
});
