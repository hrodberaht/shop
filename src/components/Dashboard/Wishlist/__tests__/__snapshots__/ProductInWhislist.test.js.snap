// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductInWishlist /> should render without crash 1`] = `
<tr>
  <td>
    ZX3
  </td>
  <td>
    400
  </td>
  <td />
  <td>
    <span>
      $
    </span>
  </td>
  <td>
    <button
      className="btn btn-danger"
      onClick={[Function]}
      type="button"
    >
      X
    </button>
  </td>
  <td>
    <AddToCartButton
      className="btn btn-primary"
      disabled={false}
      handleClickToCart={[MockFunction]}
      product={
        Object {
          "inStock": "30",
          "name": "ZX3",
          "price": 400,
          "productId": "product-1",
          "type": "Printer",
        }
      }
    />
  </td>
</tr>
`;
